From 934a7710eaaab9a5dc2a4d8e98e1b6a6efc5cf85 Mon Sep 17 00:00:00 2001
From: Michael Thalmeier <michael@thalmeier.at>
Date: Wed, 24 Jan 2018 22:02:59 +0100
Subject: [PATCH] fix build with boost 1.66

Since boost 1.66 some deprecated interfaces got removed.
Use BOOST_ASIO_ENABLE_OLD_SERVICES to re-enable some interfaces.
---
 curecoin-qt.pro     | 2 +-
 src/curecoinrpc.cpp | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/curecoin-qt.pro b/curecoin-qt.pro
index 0bde4b3..cac6054 100644
--- a/curecoin-qt.pro
+++ b/curecoin-qt.pro
@@ -2,7 +2,7 @@ TEMPLATE = app
 TARGET = curecoin-qt
 VERSION = 1.9.5.1
 INCLUDEPATH += src src/json src/qt
-DEFINES += QT_GUI BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE BOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN __NO_SYSTEM_INCLUDES
+DEFINES += QT_GUI BOOST_THREAD_USE_LIB BOOST_SPIRIT_THREADSAFE BOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN BOOST_ASIO_ENABLE_OLD_SERVICES __NO_SYSTEM_INCLUDES
 CONFIG += no_include_pwd
 QT += core gui network
 
diff --git a/src/curecoinrpc.cpp b/src/curecoinrpc.cpp
index 99853c1..5fd576e 100644
--- a/src/curecoinrpc.cpp
+++ b/src/curecoinrpc.cpp
@@ -757,7 +757,7 @@ void ThreadRPCServer2(void* parg)
 
     asio::io_service io_service;
 
-    ssl::context context(io_service, ssl::context::sslv23);
+    ssl::context context(ssl::context::sslv23);
     if (fUseSSL)
     {
         context.set_options(ssl::context::no_sslv2);
@@ -773,7 +773,7 @@ void ThreadRPCServer2(void* parg)
         else printf("ThreadRPCServer ERROR: missing server private key file %s\n", pathPKFile.string().c_str());
 
         string strCiphers = GetArg("-rpcsslciphers", "TLSv1+HIGH:!SSLv2:!aNULL:!eNULL:!AH:!3DES:@STRENGTH");
-        SSL_CTX_set_cipher_list(context.impl(), strCiphers.c_str());
+        SSL_CTX_set_cipher_list(context.native_handle(), strCiphers.c_str());
     }
 
     // Try a dual IPv6/IPv4 socket, falling back to separate IPv4 and IPv6 sockets
@@ -1067,7 +1067,7 @@ Object CallRPC(const string& strMethod, const Array& params)
     // Connect to localhost
     bool fUseSSL = GetBoolArg("-rpcssl");
     asio::io_service io_service;
-    ssl::context context(io_service, ssl::context::sslv23);
+    ssl::context context(ssl::context::sslv23);
     context.set_options(ssl::context::no_sslv2);
     asio::ssl::stream<asio::ip::tcp::socket> sslStream(io_service, context);
     SSLIOStreamDevice<asio::ip::tcp> d(sslStream, fUseSSL);
